### PR TITLE
WebGLShadowMap: Ensure to use the latest instance of WebGLObjects.

### LINKED ISSUE #17161: WebGLShadowMap not handling context loss/restore?
##### Description of the problem

We're seeing an issue where when the WebGL context is lost, WebGLShadowMap.render is dying with this call stack:
```
Uncaught TypeError: Cannot read property 'type' of undefined
    at WebGLIndexedBufferRenderer.setIndex (three.js:15584)
    at WebGLRenderer.renderBufferDirect (three.js:23376)
    at renderObject (three.js:19222)
    at renderObject (three.js:19234)
    at WebGLShadowMap.render (three.js:19064)
    at WebGLRenderer.render (three.js:23781)
    at render (webgl_shadowmap.html:406)
    at animate (webgl_shadowmap.html:378)
```
This can be reproduced in the webgl_shadowmap example. I've attached a modified version from r104 (I'm having trouble testing in Dev due to 'Strict MIME type checking for module scripts' errors) that adds a button to lose/restore the context. You might have repeat the loss/restore cycle a couple times before the error shows up.

[webgl_shadowmap.zip](https://github.com/mrdoob/three.js/files/3462630/webgl_shadowmap.zip)

##### Three.js version

- [ ] Dev 
- [x] r104

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

### PR DESCRIPTION
**Pull request by Mugen87**
Fixed #17161

### CODE REVIEW

File: src/renderers/WebGLRenderer.js [hunk 1]
```diff
@@ -304,6 +304,7 @@ function WebGLRenderer( parameters ) {
 		_this.renderLists = renderLists;
 		_this.state = state;
 		_this.info = info;
+		_this.objects = objects;
```
Comment (mrdoob) [inline src/renderers/WebGLRenderer.js]: Oh wait. Objects was not public before?

Comment (Mugen87) [reply src/renderers/WebGLRenderer.js]: Nope. Um...yeah making `objects` public is not an ideal solution. But ideally `WebGLShadows` can use the interface of `WebGLRenderer` to access the latest `WebGLObjects` instance?

Comment (Mugen87) [reply src/renderers/WebGLRenderer.js]: Or we just do something like:
```js
_this.__objects = objects;
```

Comment (WestLangley) [reply src/renderers/WebGLRenderer.js]: Regarding nomenclature, I would name it `webglObjects`.

Comment (Mugen87) [reply src/renderers/WebGLRenderer.js]: > _this.renderLists = renderLists;
> _this.state = state;
> _this.info = info;

These properties do not user the `webgl` prefix, too. So I think `objects` is more consistent to the current nomenclature of `WebGLRenderer`.

**Comment by Mugen87** [comment]
Closing in favor of #17235

**Pull request closed without merging**